lib/src/http_multipart_form_data.dart

```diff
@@ -129,8 +129,8 @@ class HttpMultipartFormData extends Stream {
       this.isText);
 
   @override
-  StreamSubscription listen(void Function(dynamic) onData,
-      {void Function() onDone, Function onError, bool cancelOnError}) {
+  StreamSubscription listen(void Function(dynamic)? onData,
```

Comment:
It feels a bit weird to allow a nullable `onData` function here, is that necessary?

Reply:
Actually, `stream.dart` defines `onData` as optional:

```dart
  StreamSubscription<T> listen(void onData(T event)?,
      {Function? onError, void onDone()?, bool? cancelOnError});
```

Reply:
weird haha, fair enough though